### PR TITLE
MODPATBLK-52

### PR DESCRIPTION
**Pull request by alexanderkurash**
Resolves [MODPATBLK-52](https://issues.folio.org/browse/MODPATBLK-52).
Implement simplified (without calendar integration) algorithm for calculating blocks related to overdue.

**Approach**
Reuse code from MODPATBLK-18, remove calendar integration.

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -13,6 +13,30 @@
...
+      "version": "7.0"
+    },
+    {
+      "id": "loan-policy-storage",
+      "version": "2.1"
+    },
+    {
+      "id": "configuration",
+      "version": "2.0"
+    },
```
Comment (alexanderkurash) [inline descriptors/ModuleDescriptor-template.json:28]: Remove this dependency

File: descriptors/ModuleDescriptor-template.json [hunk 2]
```diff
@@ -13,6 +13,30 @@
...
+    {
+      "id": "item-storage",
+      "version": "8.4"
+    },
```
Comment (alexanderkurash) [inline descriptors/ModuleDescriptor-template.json:32]: Remove this dependency

File: descriptors/ModuleDescriptor-template.json [hunk 3]
```diff
@@ -13,6 +13,30 @@
...
+    {
+      "id": "locations",
+      "version": "3.0"
+    },
```
Comment (alexanderkurash) [inline descriptors/ModuleDescriptor-template.json:36]: Check if we need this dependency

File: descriptors/ModuleDescriptor-template.json [hunk 4]
```diff
@@ -27,6 +51,17 @@
...
+          ],
+          "modulePermissions": [
+            "pubsub.event-types.post",
+            "pubsub.publishers.post",
+            "pubsub.subscribers.post",
+            "circulation-storage.loans.item.get",
+            "circulation-storage.loan-policies.item.get",
+            "configuration.entries.collection.get",
+            "inventory-storage.items.item.get",
+            "inventory-storage.locations.item.get",
```
Comment (alexanderkurash) [inline descriptors/ModuleDescriptor-template.json:63]: Check if we need these permissions

File: src/main/java/org/folio/service/OverduePeriodCalculatorService.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+          return succeededFuture(ZERO_MINUTES);
+        }
+      });
+  }
+
+  private boolean loanIsOverdue(Loan loan, DateTime systemTime) {
+    return loan.getDueDate().before(systemTime.toDate());
+  }
+
+  private Future<Integer> minutesOverdueIncludingClosedPeriods(Loan loan, DateTime systemTime) {
```
Comment (alexanderkurash) [inline src/main/java/org/folio/service/OverduePeriodCalculatorService.java:69]: Rename to calculateOverdueMinutes()

File: src/test/java/org/folio/rest/impl/AutomatedPatronBlocksAPITest.java [hunk 6]
```diff
@@ -507,6 +559,99 @@ public void updatedValuesFromConditionArePassedToResponse(TestContext context) {
...
+
+    sendRequest(USER_ID)
+      .then()
+      .statusCode(200)
+      .contentType(ContentType.JSON)
+      .body(equalTo(expectedResponse));
+  }
+
+  @Test
+  public void blockWhenLoanIsOverdue() {
```
Comment (alexanderkurash) [inline src/test/java/org/folio/rest/impl/AutomatedPatronBlocksAPITest.java:627]: Add test for loan not being overdue just because of grace period

**Review by roman-barannyk — APPROVED** [review]

**Pull request merged**